### DISCUSSION TITLE
Updates to the Gatsby Starters Library

### DISCUSSION THREAD

**Discussion by thinkybeast** [gatsbyjs/gatsby#33346] [Community]
We wanted to provide a few updates on our intended direction for the [Starters Library](http://gatsbyjs.com/starters) and hear your thoughts.

## Historical Context

When we introduced the Starters Library, it served two functions:

1. Provide templates for people to quickly start all types of Gatsby sites
2. Act as a showcase for the great sites that the community has built with Gatsby.

As of this summer, nearly 500 starters appeared in the Starters Library, most of them developed by the community.

## The Current Challenge

Since the Starters Library was first created, Gatsby has also grown enormously. With Gatsby 4, new rendering modes like DSG and SSR make more possible than ever. And alongside it, Gatsby Cloud has become the best place to build and host Gatsby sites.

These improvements have come with a tradeoff: one of the clearest pieces of feedback we've heard from developers is that it has become more difficult to find the right Starter for their next project. Here's what we've heard:

- There are a lot of Starters in the Library, but many of them are built on older versions of Gatsby, or use outdated packages.
- Some of the Starters don't build, or they build locally but not on Gatsby Cloud.
- It is not clear whether the technical implementation of each Starter is the 'optimal' approach.
- As a result, it is difficult to search through the various options in the Library and feel confident you've made the right choice.

## What We're Doing About It

Our goal is to make sure that every person that visits the Starters Library - whether they are just beginning their Gatsby journey or are kickstarting their next professional project - finds a Starter that meets their use case. We know that's an ambitious goal, given the spectrum of creative possibilities, but we're committed to pushing toward that horizon.

So, over the next few months, we're discussing some changes to the Starters Library to make it the best place to get started with a Gatsby project.

These are the improvements the Gatsby team currently has planned:

*Underway:*

- **Design changes:** We're improving the design of the Library to highlight the most common use-cases for Starters

*In Fall 2021:*

- **Starter Library Guidelines:** These guidelines will coincide with the GA release of Gatsby 4. Starters that do not meet these guidelines will not appear in the Library.
    - Must be on latest major version of Gatsby
    - Builds without error on Gatsby Cloud
    - Must have a demo deployed to Gatsby Cloud
    - Demonstrates best practices around accessibility
        - For details and resources on accessibility standards, see [https://a11y.coffee/](https://a11y.coffee/)
    - Demonstrates best practices around performance
        - For details and resources on performance, see:
            - Frontend performance: [https://web.dev/lighthouse-performance/](https://web.dev/lighthouse-performance/)
            - Gatsby build performance: [https://www.gatsbyjs.com/docs/how-to/performance/improving-build-performance/](https://www.gatsbyjs.com/docs/how-to/performance/improving-build-performance/)
    - Includes documentation on installation and contribution

The most noticeable result of this change is that fewer Starters will appear in the Library, but people will be able to select any given Starter with confidence that it will work without initial maintenance.

*In Fall 2021 and beyond:*

- **Feedback widget:** Similar to the feedback widget on our Documentation pages, we plan to add a similar widget to our Starters to understand what's working and what's missing from the Library.
- **Re-opening community submissions:** Community submissions to the Starters Library were paused in Summer 2020. We're looking forward to opening up a new process for submitting a Starter for addition to the Library. Although we don't have details yet, we want to develop this process along with the community, so leave a comment with what you'd like that process to look like! With all the new hotness in Gatsby v4, we can't wait to see the Starters you build. 💜

## What Do You Think?

We know this decision impacts you, our wonderful open-source community, and so we'd love to hear your thoughts on the changes we've planned so far.

- Do these solutions we've outlined feel helpful? Which ones feel most impactful? Which ones feel least impactful?
- Do you have any suggestions for additional or better solutions?
- What else have we missed or not considered?

**Comment by herecydev** [comment]
What consitutes "Demonstrates best practices"? One of the thing I've commonly seen before is how "Gatsby fails to meet X on lighthouse scores" or is "slow for x", which from my experience is simply untrue. I always use the `gatsby-starter-default` as it demonstrates perfect scores.

Would it be unreasonable to only permit 90+ (or whatever) on each score? From my perspective, a starter should always start "perfect" and it's the user's additions that have the _potential_ to reduce those scores.

**Reply by thinkybeast** [reply]
> From my perspective, a starter should always start "perfect" and it's the user's additions that have the potential to reduce those scores.

That aligns with how the team has been thinking about this too. People should have a reasonable expectation that every Starter that appears in the Library will be performant and accessible out of the box. Given that Lighthouse scores have become a standard way to evaluate this, people should therefore expect every Starter in the Library to have [a "good" score (90-100) ](https://web.dev/performance-scoring/#color-coding) in these categories.

**Comment by justinphilpott** [comment]
Perhaps there would be value in creating a small set of example (made "in house") starters that demonstrate what is referred to as "best practice" for the common kinds of site that one might wish to create. Lets say "blog", "brochure", "portfolio", "store" and perhaps a couple of others... These could be simply or generically themed (as past in-house themes have been) and would serve two purposes:
1. They would be a start point for community contrib to build upon said "best practice", demonstrating good usage of latest common plugins.
2. They would be a trusted generic site "template" for consumers to select of the shelf to kick of projects before the new "starters" section is fully filled with new community contrib.

**Reply by thinkybeast** [reply]
@justinphilpott Great points. We've recently had some discussions along the same lines. I'd expect to see more around this early next year :)

**Comment by cabinman777** [comment]
For meI would just sayI liked the big library in Gatsby 3 . I was able to create some really cool sites and a variety of them to the pleasure of clients. In my opinion a big library of starters is the way to go

**Comment by Sepci0** [comment]
What I dislike most is, there are 

> nearly 500 starters (...) in the Starters Library

and now they are all unavailable. 
I loved to go through them for inspiration, and find one to "start" with. 
I understand that you want to have v4 having great quality, but why did you deleted the access to all the old stuff with information

> The *name* has become deprecated, please choose a starter from the starters below that are up to date and use best practices!

[[source](https://www.gatsbyjs.com/starters/?deprecated-starter=%25name%25)]
Now all the 500 starters that ppl made, all of this work, all is inaccessible (more like - hidden).
All my saved links to decent starters are now useless, and I have to dig out the creator name to find something on their personal github. 
I know with the time there will be more starters for v4, but for now starters are unusable for me.

**Reply by jxnblk** [reply]
Hi @Sepci0, the starters "removed" from the site are still available, since they are hosted on GitHub. It's not perfect, but you can find a list of the starters here: https://github.com/gatsbyjs/gatsby/blob/gatsby%402.23.0/docs/starters.yml

We really want to make sure that new users get started with the latest and greatest version of Gatsby and have the best experience, and we don't want to put the work of upgrading to the latest version on those folks. As we get more starters updated to v4, I'd expect to see more starters in the new library over time.

**Comment by konstantinruge** [comment]
I see the point and agree in general that starters should fulfill a certain baseline of quality. So, I see where you are coming from 👍  

I myself maintain a starter that is currently unavailable. Are there updated docs on how to submit starters to the new library? I upgraded the project to v4 already. What else do I need to do?

**Comment by darkowic** [comment]
It's a shame... I wanted to kick off a new project with some nice templates and the selection has become so limited :/ Why can't you mark the old templates as outdated or sth **and give people some time to migrate**? Or at least have a separate page with the old templates. And no, the huge YAML is not what we are looking for...

I understand the quality concerns and it's really appreciated but for now, the starters page is unusable... I can not send people to the page to see different possibilities and for inspiration anymore.

**Reply by RookTKO** [reply]
Was there not a period of time that everyone was alerted to the transfer/ elimination of the old templates? I'm curious to this, because I feel like this should have happened.

**Comment by thinkybeast** [comment]
Hearing a couple of distinct themes so far:

1. You'd like a way to view the pre-version 4 starters.
- Understood! We've made the previous version of the Starters library available at [https://v3.gatsbyjs.com/starters](https://v3.gatsbyjs.com/starters). Given your input, we're also discussing new ways to view older versions of community-built starters in the future.

2. You'd like clarity on a submission process for the current library.
- We're working on a new submission process/guidelines, and aim to have that available before the end of the year. In the meantime, if you comment on this thread with the repo you'd like to add, we'll follow up with you directly.

**Reply by konstantinruge** [reply]
Thanks for the response! For me, it is this repo in specific that I would like to add to the starters library: [Gatsby Starter Portfolio Minimal Theme](https://github.com/konstantinmuenster/gatsby-starter-portfolio-minimal-theme)

**Reply by donaldboulton** [reply]
I would like to add a new Gatsby Starter. Gatsby v4.1.4  to the Library. Gatsby Starter Basic Instructions, [Repo](https://github.com/donaldboulton/gatsby-starter-basic-instructions) Hosted on Gatsby Cloud at [Basic Instructions](https://gatsbystarterbasicinstructions.gatsbyjs.io/)

Featuring the Latest Gatsby 4.1.4 theming with Theme-UI. Full text and image search, tags, with frontmatter featuredImage's and markdown images using gatsby-plugin-image for a picture srcSet.

**Reply by jpedroschmitz** [reply]
I'd like to add https://github.com/jpedroschmitz/gatsby-starter-ts in the library.

**Reply by LekoArts** [reply]
@konstantinmuenster @jpedroschmitz you'll need to adapt your projects to adhere to the mentioned "Starter Library Guidelines" in the start post.

**Reply by flogy** [reply]
I'd like to update the currently deprecated https://github.com/flogy/gatsby-starter-capacitor (just updated to Gatsby v4 and deployed to Gatsby Cloud) 🚀

**Reply by jpedroschmitz** [reply]
@LekoArts, I deployed the starter on Gatsby Cloud. Let me know if there's anything else that I should be aware of. Thanks!

**Reply by TheeMattOliver** [reply]
Proposing "Gatsby Theme UI Starter".

- Must be on latest major version of Gatsby ✅  

 Gatsby 4.3.0

- Builds without error on Gatsby Cloud ✅ 

https://build-b8f81337-a8b5-499d-bbfe-c09ee94294d9.gtsb.io/

- Must have a demo deployed to Gatsby Cloud ✅  

https://gatsbythemeuistarter.gatsbyjs.io/

 - Includes documentation on installation and contribution ✅ 

https://github.com/TheeMattOliver/gatsby-theme-ui-boilerplate/blob/main/CONTRIBUTING.md

- Demonstrates best practices around accessibility ✅ 

- Demonstrates best practices around performance ✅  

<img width="1324" alt="Screen Shot 2021-12-11 at 4 40 42 PM" src="https://user-images.githubusercontent.com/9900326/145693791-60ae6df2-8d53-44e8-a8a2-1e4d48103883.png">

Pls let me know next steps, thanks.

**Reply by flogy** [reply]
When will the starters linked in this discussion be added to the library?

**Reply by thinkybeast** [reply]
@flogy We're working on revamping how we want to display starters given the feedback in this thread, as well as finalizing a new submission process. We'll have more news in the first couple weeks of January.

**Reply by thinkybeast** [reply]
@flogy Added more details in [the comment below!](https://github.com/gatsbyjs/gatsby/discussions/33346#discussioncomment-1999947)

**Comment by izackwu** [comment]
It's too bad that all those starters have been taken down now. A lot of people like me just want to browse the gallery and pick some nice-looking starter to use immediately, rather than develop based on them following whatever best practices are. 

I understand that we want to clarify the scope of starters and thus remove those outdated ones, but it's really necessary to have such an awesome-list of website templates made with Gatsby.

**Comment by ChristineTham** [comment]
Hi, I would like to submit a starter for consideration.
[Hello Gatsby Starter](https://github.com/hellotham/hello-gatsby-starter)

I think I have met all the requirements (but please check) - it builds with the latest version of all packages, with a demo deployed to Gatsby Cloud:
- [hello-gatsby-starter.hellotham.com](https://hello-gatsby-starter.hellotham.com)
- [hellogatsbystarter1.gatsbyjs.io](https://hellogatsbystarter1.gatsbyjs.io)

Is there anything else I need to do?

**Comment by donaldboulton** [comment]
I have another Starter for Consideration [Deployed to Netlify as Dimension V4](https://gatsby-starter-dimension-v4.netlify.app/). [Repo](https://github.com/donaldboulton/gatsby-starter-dimension-v4)

I am Poor and Already Have A Starter on Gatsby Cloud and I Get only 1 free site, so to Netlify. Has all the Gatsby cloud prerequisites for Gatsby Cloud

**Comment by thundermiracle** [comment]
IMO, what the community needs is no only the `perfect` starters, but how to combine / use libraries, how to implement `some functions` like i18n as well. We cannot just kill those `how to` starters although some of them are not that elegant.
It's OK to separate the recommended official ones with the `normal` ones, but filtering by functions is really helpful.
Implementing all these `how to` starters one by one by core members is also acceptable.

What I'm talking about is just like examples in Next.js:
https://github.com/vercel/next.js/tree/canary/examples

Want to use `material-ui` with Next.js? Simple, just access the examples folder and open the `with-material-ui` folder.

**Comment by roncham** [comment]
It's a shame to lose 500 starters. I'm glad you created the link to the V3 starters to make them available. I was shocked when I saw such a limited # of starters.
An in house gatsby-woocommerce starter w/ a blog should be available with all the headless cms talk.
I agree every starter should score 90+ out of the box.

**Comment by sungik-choi** [comment]
What are the criteria for being listed in the library? In the case of blog starters, there are not many promotional channels, so I hope the community will be activated soon and it will be a place where can share the starter more freely. 

https://github.com/sungik-choi/gatsby-starter-apple

I recently updated it to v4, so it would be nice to have a place to share it quickly. The wait is too long...

**Comment by thinkybeast** [comment]
Hey all, providing an update on how to have your starter included in the Library.

We recently re-opened submissions to the library! You can find a form to submit your starter(s), along with the criteria for being added, at this link: https://www.gatsbyjs.com/starters/submissions

Our plan is to review and add the first batch of submissions we receive toward the second half of February, and then review new submissions on a rolling basis, every month or so. 

For those of you who have dropped your starter details in this thread, we'll follow up with you. In some cases, we'll ask you to re-submit your details via the submission form, as it asks for additional information not provided in the thread. 

Thanks again!
